modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/SegmentResourceImpl.java

```diff
@@ -131,14 +134,27 @@ private Segment _toSegment(SegmentsEntry segmentsEntry) {
 		return new Segment() {
 			{
 				active = segmentsEntry.isActive();
-				criteria = segmentsEntry.getCriteria();
+				criteria = segmentsEntry.getCriteriaObj();
 				dateCreated = segmentsEntry.getCreateDate();
 				dateModified = segmentsEntry.getModifiedDate();
 				id = segmentsEntry.getSegmentsEntryId();
 				name = segmentsEntry.getName(
 					segmentsEntry.getDefaultLanguageId());
 				siteId = segmentsEntry.getGroupId();
-				source = segmentsEntry.getSource();
+
+				setSource(
+					() -> {
+						if (Objects.equals(
```

Comment:
It might be clearer to use StringUtils.equals to better undestand that the segmentsEntry.getSource is a String